src/Analysis/Engine/Impl/Interpreter/Ast/NameLookupContext.cs

```diff
@@ -346,27 +354,61 @@ sealed class NameLookupContext {
             }
 
             var m = GetValueFromExpression(expr.Target);
-            if (m is IPythonType type) {
-                if (type.TypeId == BuiltinTypeId.Type && type == Interpreter.GetBuiltinType(BuiltinTypeId.Type) && expr.Args.Count >= 1) {
-                    var aType = GetTypeFromValue(GetValueFromExpression(expr.Args[0].Expression));
-                    if (aType != null) {
-                        return aType;
+            IMember value = null;
+            switch (m) {
+                case IPythonType type:
+                    if (type.TypeId == BuiltinTypeId.Type && type == Interpreter.GetBuiltinType(BuiltinTypeId.Type) && expr.Args.Count >= 1) {
+                        value = GetTypeFromValue(GetValueFromExpression(expr.Args[0].Expression, options));
+                    } else {
+                        value = new AstPythonConstant(type, GetLoc(expr));
                     }
-                }
-                return new AstPythonConstant(type, GetLoc(expr));
+                    break;
+                default:
+                    value = GetValueFromPropertyOrFunction(m, expr);
+                    break;
             }
+            if (value == null) {
+                _log?.Log(TraceLevel.Verbose, "UnknownCallable", expr.Target.ToCodeString(Ast).Trim());
+                return new AstPythonConstant(_unknownType, GetLoc(expr));
+            }
+            return value;
+        }
+
+        private IMember GetValueFromPropertyOrFunction(IMember fn, Expression expr) {
+            switch (fn) {
+                case IPythonBoundFunction bf:
```

Comment:
If we move type check into `case`:
```cs
case IPythonBoundFunction bf when bf.Function as IPythonFunction2 f: 
    GetValueFromFunction(f, expr);
```
we can get rid of null check in the `GetValueFromFunction`.

Reply:
Sure